frontend/src/contexts/AuthContext.tsx

```diff
@@ -35,10 +35,7 @@ export function AuthProvider({ children }: ProviderProps) {
     return unsubscribe;
   }, []);
 
-  const value = useMemo(
-    () => ({ currentUser, signedIn: currentUser !== undefined, loading }),
-    [currentUser],
-  );
+  const value = useMemo(() => ({ currentUser, signedIn: !currentUser, loading }), [currentUser]);
```

Comment:
Think the issue with sign in could be here, as I believe you're setting signedIn to true if the currentUser doesn't exist and false if it does.

Reply:
```suggestion
  const value = useMemo(() => ({ currentUser, signedIn: !!currentUser, loading }), [currentUser]);
```

Ah good catch. This should fix it